File: examples/python/pink-fork.py
#!/usr/bin/env python
# coding: utf-8
# vim: set sw=4 ts=4 sts=4 et :

"""\
An example demonstrating the tracing fork
"""

from __future__ import print_function

import pinktrace.fork
import pinktrace.trace

pid = pinktrace.fork.fork(pinktrace.trace.OPTION_ALL)
if not pid: # child
    print("hello world")
else: # parent
    # At this point the child has been stopped for tracing and stopped itself
    # using SIGSTOP. We don't do anything interesting for this example.

    # Kill the child, the print function will never be called.
    pinktrace.trace.kill(pid)

File: examples/ruby/pink-fork.rb
#!/usr/bin/env ruby
# coding: utf-8
# vim: set sw=2 sts=2 et fenc=utf-8 :

=begin
An example demonstrating the tracing fork
=end

require 'PinkTrace'

pid = PinkTrace::Fork.fork(PinkTrace::Trace::OPTION_ALL) do
  puts "hello world"
end

# At this point the child has been stopped for tracing and stopped itself
# using SIGSTOP. We don't do anything interesting for this example.

# Kill the child, the puts function will never be called.
PinkTrace::Trace.kill pid
